prometheus/testutil/testutil.go

```diff
@@ -108,6 +108,34 @@ func ToFloat64(c prometheus.Collector) float64 {
 	panic(fmt.Errorf("collected a non-gauge/counter/untyped metric: %s", pb))
 }
 
+// MetricCount collects all Metrics from the provided Collector and counts them.
+//
+// Typical use case is when you want to assert certain number of series of given collector after
+// some operation.
```

Comment:
Suggestion: This can be used to assert the number of metrics collected by a given collector after certain operations.